src/Core/Framework/Api/Controller/SalesChannelProxyController.php

```diff
@@ -412,18 +412,24 @@ private function updateCustomerToContext(string $customerId, SalesChannelContext
         $this->eventDispatcher->dispatch($event);
     }
 
-    private function persistPermissions(Request $request): void
+    private function persistPermissions(Request $request, SalesChannelContext $salesChannelContext): void
     {
-        $contextToken = $this->getContextToken($request);
+        $contextToken = $salesChannelContext->getToken();
 
-        $salesChannelId = (string) $request->request->get('salesChannelId');
+        $salesChannelId = $salesChannelContext->getSalesChannelId();
 
         $payload = $this->contextPersister->load($contextToken, $salesChannelId);
+        $requestPermissions = $request->get(SalesChannelContextService::PERMISSIONS);
 
-        if (!\in_array(SalesChannelContextService::PERMISSIONS, $payload, true)) {
-            $payload[SalesChannelContextService::PERMISSIONS] = self::ADMIN_ORDER_PERMISSIONS;
```

Comment:
The API caller expects that the `ADMIN_ORDER_PERMISSIONS` is added. With your change is it not so.

Can you change the admin call, that it requests this permission and ensure in Controller that this permission is still always added?

Reply:
The custom permissions are only set if you define them explicitly. If the request does not contain any permissions, the behavior is not changed